perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/UsersManagerBlImpl.java

```diff
@@ -624,6 +624,7 @@ public UserExtSource updateUserExtSource(PerunSession sess, UserExtSource userEx
 
 	@Override
 	public void updateUserExtSourceLastAccess(PerunSession sess, UserExtSource userExtSource) throws InternalErrorException {
+		getPerunBl().getAuditer().log(sess, new UserExtSourceUpdated(userExtSource));
```

Comment:
You should always generate message after the process is done. You can prevent unexpected issues.

Reply:
Fixed, also in standard ``updateUserExtSource()`` method.